primary/src/grpc_server/configuration.rs

```diff
@@ -146,4 +154,15 @@ impl Configuration for NarwhalConfiguration {
 
         Ok(Response::new(Empty {}))
     }
+
+    async fn get_primary_address(
+        &self,
+        _request: Request<Empty>,
+    ) -> Result<Response<GetPrimaryAddressResponse>, Status> {
```

Comment:
Since this always succeed, does it need to return a `Result`?

Reply:
tonic requires `Result` here

```
 note: expected fn pointer `fn(&'life0 NarwhalConfiguration, tonic::Request<_>) -> Pin<Box<(dyn futures::Future<Output = Result<tonic::Response<GetPrimaryAddressResponse>, Status>> + std::marker::Send + 'async_trait)>>`
 ```